src/modules/sni/watcher.cpp

```diff
@@ -23,6 +22,7 @@ Watcher::~Watcher() {
     g_slist_free_full(items_, gfWatchFree);
     items_ = nullptr;
   }
+  Gio::DBus::unown_name(bus_name_id_);
```

Comment:
It's not already the case when it's destroyed?

Reply:
Where is it destroyed ? I worked on the assumption that (bus_name_id_) is only an int, and therefore RAII does not apply.

Reply:
Ok you're right, I assumed that Gio::DBus::own_name will do it, but no.